Portmap support for dual-stack (#4361)
cri-o/cri-o: internal/hostport/README.md

![CRI-O logo](https://github.com/cri-o/cri-o/blob/master/logo/crio-logo.svg?raw=true)
# Host port manager

CRIO-O was vendoring the host port manager implementation from Kubernetes.
Kubernetes started the deprecation of the dockershim code and there is a risk that the upstream code will not

> receive any new patches or fixes.

Reviewer: also might want a rewording

Author: done

![CRI-O logo](https://github.com/cri-o/cri-o/blob/master/logo/crio-logo.svg?raw=true)
# Host Port Manager

> The Host Port Managers map ports from the Host to the specified Container Port.

Reviewer 2: ```suggestion
The Host Port Manager maps ports from the Host to the specified Container Port.
```

It can also specify the Host IP that should be mapped to the container IP.

> For security reasons, it also open a socket in the host to avoid any process in the system

Reviewer 2: ```suggestion
For security reasons, it also opens a socket in the host to avoid any process in the system
```

> can interfere with the traffic that is being port mapped.

Reviewer 2: ```suggestion
that could interfere with the traffic that is being port mapped.
```

![hostport](HostPort.jpg "Container Host Port")

This code is copied from the kubernetes/kubernetes repo

> https://github.com/kubernetes/kubernetes/tree/41533ecec8966636554faba44368a36d5d6c5347/pkg/kubelet/dockershim/network/hostport

Reviewer 2: ```suggestion
https://github.com/kubernetes/kubernetes/tree/41533ecec8966636554faba44368a36d5d6c5347/pkg/kubelet/dockershim/network/hostport.
```

> since Kubernetes started the deprecation of that part of the code in

Reviewer 2: ```suggestion
Since Kubernetes started the deprecation of that part of the code in
```

> https://github.com/kubernetes/kubernetes/pull/94624 and is missing some fixes and the features required to operate

Reviewer 2: ```suggestion
https://github.com/kubernetes/kubernetes/pull/94624, it is missing some fixes and the features required to operate
```

correctly in dual stack environments.

> It has a fix where the host port manager always open a socket in all addresses, instead of leveraging the HostIP field

Reviewer 2: ```suggestion
 The kubernetes/kubernetes repo has a fix where the host port manager always opens a socket in all addresses, instead of leveraging the HostIP field:
```

> # Host port manager

Reviewer 2: Do we usually include the CRI-O logo in the top of the README.md files?  If not, we should for context.

Author: added

Kubernetes started the deprecation of the dockershim code and there is a risk that the upstream code will not
receive any new patches or fixes.

> This package come from https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/dockershim/network/hostport

Reviewer 2: ```suggestion
This package comes from https://github.com/kubernetes/kubernetes/tree/master/pkg/kubelet/dockershim/network/hostport
```
